deployer/src/handlers/mod.rs

```diff
@@ -609,6 +609,8 @@ pub async fn start_deployment(
     CustomErrorPath((project_name, deployment_id)): CustomErrorPath<(String, Uuid)>,
 ) -> Result<()> {
     if let Some(deployment) = persistence.get_runnable_deployment(&deployment_id).await? {
+        let account_name = claim.sub.replace('|', "-");
```

Comment:
Do we have account names with `|`?

Reply:
This `sub` field comes with `|` in them. That's how Auth0 stores users, not sure why though.

Reply:
Just took a look at this, `sub` is indeed coming from auth0, and `|` is the delimiter they use. It goes like: `{social_provider|{unique_id}}.

Reply:
Hmm, maybe I am missing something, but the claim is created here: https://github.com/shuttle-hq/shuttle/blob/main/auth/src/api/handlers.rs#L130-L142, and the sub is created from the account name that is stored in `auth`, and they are stored as `github-<id>` (at least all the ones in the staging database). Have you seen the format you speak of in your testing?

Reply:
Not sure where it's set in backend, but it always comes with `|` from auth0.

When user first registers on Shuttle using Auth0 Github connection, this piece of code is run:

```
const shuttlified = session.user.sub.replace('|', '-');
// At this point no calls have been made to our backend, `|` is coming from Auth0.

// We then create new user using this replaced version.
const user = await Shuttle.create_user(shuttlified);

// Then we set user info to session and that's basically it
session.user.api_key = user.key;
session.user.userId = user.name;
session.user.account_tier = user.account_tier;
session.user.subscription_id = user.subscription_id;
```

Reply:
Auth0 probably calls `/auth/session` endpoint suring it's oauth flow with account id having `|`.

Reply:
Ah, I see. So the claim subject will always be in the `github-<id>` format. Auth0 is not involved in the creation of the JWT, auth0 is just involved in the creation of a user, and from your codeblock it's clear that we change it to `github-<id>` before we send it to the backend, where it is persisted in the auth state. When the claim is created, it gets the account name from this state.

Reply:
We should be good to remove this line then.